apps/bluetooth/js/modules/transfer_manager.js

```diff
@@ -0,0 +1,372 @@
+            Debug('getFile succeed & file is sending...');
+            this._sendFileThenCheckTasksComplete(targetDeviceAddress, 
+                                                 getRequest.result);
+          };
+
+          getRequest.onerror = () => {
+            Debug('getFile failed so that blob is sending without filename ' +
+                  getRequest.error && getRequest.error.name);
+            this._sendFileThenCheckTasksComplete(targetDeviceAddress, blob);
+          };
+        } else {
+          // The blob does not have name and filepath,
+          // pass the blob without name to send file directly.
+          Debug('no filepath to get from device storage ' +
+                'so that blob is sending without filename');
+          this._sendFileThenCheckTasksComplete(targetDeviceAddress, blob);
+        }
+      });
+    },
+
+    /**
+     * Given address, blob(file) to send file via platform.
+     *
+     * @access private
+     * @memberOf TransferManager
+     * @param {String} address - address of target device
+     * @param {Object} blob - blob(file) to send
+     */
+    _sendFileThenCheckTasksComplete: 
+    function bttm__sendFileThenCheckTasksComplete(address, blob) {
+      BtContext.sendFile(address, blob);
+      this._checkTasksComplete();
+    },
+
+    /**
+     * Given address, blob(file) to send file via platform.
+     *
+     * @access private
+     * @memberOf TransferManager
+     */
+    _checkTasksComplete: function bttm__checkTasksComplete() {
+      if (--this._numberOfTasks === 0) {
+        // Post result with reason then end the activity service.
+        this._endTransferWithReason('transferred');
+      }
+    },
+
+    /**
+     * While a user cancel selection, let's post result then end the activity
+     * service.
+     *
+     * @access private
+     * @memberOf TransferManager
+     * @param {Object} event
+     * @param {String} event.type - type of event name
+     */
+    _onCancelSelection: function bttm__onCancelSelection(event) {
+      Debug('_onCancelSelection(): cancel from user decision');
+      // Post error with reason then end the activity service.
+      this._endTransferWithReason('cancelled');
+    },
+
+    /**
+     * A function is a role producer.
+     * Bluetooth app produce one message for each sending file request.
+     *
+     * @access private
+     * @memberOf TransferManager
+     */
+    _produceSendingFilesSchedule: function bttm__produceSendingFilesSchedule() {
+      // Construct a object to contain info of sending files schedule. 
+      // And the info will be posted message to system app.
+      // The result of send files will be displayed on notification.
+      var sendingFilesSchedule = {
+        numberOfFiles: this._activity.source.data.blobs.length,
+        numSuccessful: 0,
+        numUnsuccessful: 0
+      };
+      // Post message to system app for sending files in queue
+      this._postMessageToSystemApp(sendingFilesSchedule);
```

Comment:
Let's keep the function simple by only generating the schedule based on the blob length that passed from outside. It also improves readability if the code looks like:
```
var schedule = this._produceSendingFilesSchedule(this._activity.source.data.blobs.length);
this._postMessageToSystemApp(schedule);
```